packages/reports/src/reportBuilder/reportBuilder.ts

```diff
@@ -0,0 +1,41 @@
+import { Aggregator } from '../aggregator';
+import { FetchReportQuery } from '../routes/fetchReport';
+import { fetch, FetchResponse } from './fetch';
+import { buildTransform } from './transform';
+
+export type FieldValue = string | number | boolean | undefined;
+export const isValidFieldValue = (fieldValue: any): fieldValue is FieldValue => {
+  return (
+    typeof fieldValue === 'number' ||
+    typeof fieldValue === 'string' ||
+    typeof fieldValue === 'boolean' ||
+    fieldValue === undefined
+  );
+};
+
+export interface Row {
+  [field: string]: FieldValue;
+}
+
+export interface BuildReport {
+  results: Row[];
+}
+
+export class ReportBuilder {
+  readonly report;
+  readonly aggregator: Aggregator;
+  readonly query: FetchReportQuery;
+
+  constructor(report, aggregator: Aggregator, query: FetchReportQuery) {
+    this.report = report;
+    this.aggregator = aggregator;
+    this.query = query;
+  }
+
+  build = async (): Promise<BuildReport> => {
+    const data: FetchResponse = await fetch(this.report.config.fetch, this.aggregator, this.query);
+    const builtTransform = buildTransform(this.report.config.transform);
```

Comment:
So this pattern of building a transform function from the config is one I quite like and would like to generalise it to `fetch` and `output` when they are a bit more fleshed out.

The idea is that each step of the process is responsible for building a function from the config it's given and in that process we validate that config. Typescript forces us to validate the config anyway, and by doing it as a separate step to running the transform, we can validate the user's config at the time of saving it.